app/views/admin/secrets/edit.html.erb

```diff
@@ -1,16 +1,46 @@
+<%= javascript_include_tag "/deploy_groups.js" %>
 <% id = params[:id] %>
 <h1><%= id ? "Edit #{id}" : "New Secret" %></h1>
 
+<div id="deployGroupList" style="display: none;" imbededData="<%= DeployGroup.all.map { |g| {g.environment.permalink => g.permalink }}.to_json %>" class="hidden"></div>
+
 <section>
   <% url = (id ? admin_secret_path(params[:id]) : admin_secrets_path) %>
   <% method = (id ? :put : :post) %>
-  <% project_permalink, key = id.to_s.split('/', 2) %>
+  <% parse_environment_permalink = SecretStorage.parse_secret_key(id, :environment) %>
+  <% parse_project_permalink, key = SecretStorage.parse_secret_key(id, :project) %>
+  <% parse_deploy_group_permalink = SecretStorage.parse_secret_key(id, :deploy_group) %>
+  <% key = id.to_s %>
   <%= form_tag url, class: "form-horizontal", method: method do %>
     <fieldset>
+
+      <div class="form-group">
+        <%= label_tag 'secret[environment_permalink]', "Environment", class: "col-lg-2 control-label" %>
+        <div class="col-lg-4">
+          <%= select_tag 'secret[environment_permalink]', 
+            options_for_select(@environment_permalinks,
+              parse_environment_permalink), class: "form-control",
```

Comment:
makes it look like class etc are part of the `options_for_select` call ... how about:

```
options_for_select(@environment_permalinks, parse_environment_permalink),
class: "form-control",include_blank: true, disabled: .... 
```